### DISCUSSION TITLE
Why Prefer Alpine.data() To Global Alpine Function Data Providers?

### DISCUSSION THREAD

**Discussion by philc-d** [alpinejs/alpine#2172] [1. Help] [answered]
The V3 documentation states that Alpine.data() should be preferred, but I can't find any explanation why. What are the benefits of Alpine.data() compared to global component functions?

**Comment by danddanddand** [comment, marked as answer]
There isn't any advantage other than very slightly polluting global scope less.

In our project we choose not to use Alpine.data. We use TS with Snowpack for development and Webpack for production. So all the alpine components, etc live in that library scope. 

So I'd say it's preference thing at this point. If you don't want to pollute global scope and want short named references to your component code, Alpine.data is decent solution.

**Reply by bttger** [reply]
But why is the migration guide saying following?

> The following 2 APIs will still work in V3, but are considered deprecated and are likely to be removed at some point in the future.
> [Prefer Alpine.data() to global Alpine function data providers](https://alpinejs.dev/upgrade-guide#alpine-data-instead-of-global-functions)

I find a function data provider much more readable and apparent what it does.
